mm2src/lp_swap/maker_swap.rs

```diff
@@ -587,6 +568,23 @@ impl MakerSwap {
             ]));
         }
 
+        let f = self.maker_coin.check_if_my_payment_completed(
+            &unwrap!(self.r().maker_payment.clone()).tx_hex,
+            self.r().data.maker_payment_lock as u32,
+            &*self.r().other_persistent_pub,
+            &*dhash160(&self.r().data.secret.0),
+        );
+        if let Err(err) = f.compat().await {
+            // Note the error may caused by internet problems.
```

Comment:
Going to refund on a first network error is too fast here, can you please add at least several retries here or maybe even indicate that error is caused by network problem?
Please pick something that is faster to implement and then you can consider refactoring to something more complicated during Native mode support implementation.

Reply:
You are right, it's too fast to refund on first error. I've added several retries, but it will be refactored in the next Qrc20 iteration.